src/ckb-daemon/structures.h

```diff
@@ -247,7 +248,7 @@ typedef struct {
     // Color dithering in use
     char dither;
     // Flag to check, if large macros should be sent delayed
-    char delay;
+    uint delay;
```

Comment:
OK, thanks.

I have done some testing today. First I merged your d6aa207 and my macrotime0.1.
Then I created a new G-macro (typing very, very slowly):
`+lshift,+d=26161,-d=37140,-lshift=142738,+a=89017,-a=144970,+s=92969,-s=50021,+space=1996,-space,+w=82943,-w=49985,+i=95034,-i=78998,+r=61994,-r=91999,+d=81007,-d=51698,+space,-space,+e=100986,-e=35010,+i=86196,-i=53792,+n=15992,-n=46003,+space=3206,-space=2798,+lshift=76004,+t=9013,-t=3990,-lshift=32303,+e=61019,-e=6009,+x=86265,-x=49755,+t=1956,-t=4440,+space=97006,-space=3994,+m=97974,+i=19981,-m=6008,-i=1998,+t=630,-t=7036,+space=8959,-space=65303,+lshift=7099,+d,-lshift,-d,+e,-e,+l,+a,-l=31996,+z=56249,-a=58814,-z=58948,+enter=69214,-enter`

That gives a string "Das wird ein Text mit Delay", which means "That gets a text with delay<enter>". Trying this macro runs well.
But if I repeat pressing that G-key (10, 20 times) while the macro is just running, I get sometimes a disconnect of the KB. The daemon is killed and restartet from init.

## The logging says something about usb disconnected (there is an unformmated file at the end of this post):

`22.08.2016 20:14:59 Mainfrix    kernel  [20612.403540] input: ckb1: Corsair K95 RGB Gaming Keyboard as /devices/virtual/input/input117
22.08.2016 20:16:42 Mainfrix    kernel  [20715.383813] usb 1-3: USB disconnect, device number 46
22.08.2016 20:16:42 Mainfrix    kernel  [20715.657524] usb 1-3: new full-speed USB device number 47 using xhci_hcd
22.08.2016 20:16:44 Mainfrix    acpid   input device has been disconnected, fd 17
22.08.2016 20:16:44 Mainfrix    acpid   input device has been disconnected, fd 25
22.08.2016 20:16:47 Mainfrix    kernel  [20720.793068] usb 1-3: unable to read config index 0 descriptor/all
22.08.2016 20:16:47 Mainfrix    kernel  [20720.793073] usb 1-3: can't read configurations, error -110
22.08.2016 20:16:47 Mainfrix    kernel  [20720.905202] usb 1-3: new full-speed USB device number 48 using xhci_hcd
22.08.2016 20:16:48 Mainfrix    kernel  [20721.034919] usb 1-3: New USB device found, idVendor=1b1c, idProduct=1b11
22.08.2016 20:16:48 Mainfrix    kernel  [20721.034923] usb 1-3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
22.08.2016 20:16:48 Mainfrix    kernel  [20721.034925] usb 1-3: Product: Corsair K95 RGB Gaming Keyboard
22.08.2016 20:16:48 Mainfrix    kernel  [20721.034926] usb 1-3: Manufacturer: Corsair
22.08.2016 20:16:48 Mainfrix    kernel  [20721.034928] usb 1-3: SerialNumber: 0F022014AE3B9406537275B6F5001945
22.08.2016 20:16:48 Mainfrix    kernel  [20721.036220] input: Corsair Corsair K95 RGB Gaming Keyboard  as /devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3:1.0/0003:1B1C:1B11.0051/input/input118
22.08.2016 20:16:48 Mainfrix    kernel  [20721.089793] hid-generic 0003:1B1C:1B11.0051: input,hidraw2: USB HID v1.11 Keyboard [Corsair Corsair K95 RGB Gaming Keyboard ] on usb-0000:00:14.0-3/input0
22.08.2016 20:16:48 Mainfrix    kernel  [20721.090869] input: Corsair Corsair K95 RGB Gaming Keyboard  as /devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3:1.1/0003:1B1C:1B11.0052/input/input119
22.08.2016 20:16:48 Mainfrix    kernel  [20721.145683] hid-generic 0003:1B1C:1B11.0052: input,hidraw3: USB HID v1.11 Keyboard [Corsair Corsair K95 RGB Gaming Keyboard ] on usb-0000:00:14.0-3/input1
22.08.2016 20:16:48 Mainfrix    kernel  [20721.146367] hid-generic 0003:1B1C:1B11.0053: hiddev0,hidraw4: USB HID v1.11 Device [Corsair Corsair K95 RGB Gaming Keyboard ] on usb-0000:00:14.0-3/input2
22.08.2016 20:16:48 Mainfrix    kernel  [20721.147014] hid-generic 0003:1B1C:1B11.0054: hiddev0,hidraw5: USB HID v1.11 Device [Corsair Corsair K95 RGB Gaming Keyboard ] on usb-0000:00:14.0-3/input3
22.08.2016 20:16:48 Mainfrix    mtp-probe   checking bus 1, device 48: "/sys/devices/pci0000:00/0000:00:14.0/usb1/1-3"
22.08.2016 20:16:48 Mainfrix    mtp-probe   bus: 1, device: 48 was not an MTP device
22.08.2016 20:16:48 Mainfrix    acpid   input device has been disconnected, fd 17
22.08.2016 20:16:48 Mainfrix    acpid   input device has been disconnected, fd 25
22.08.2016 20:16:48 Mainfrix    kernel  [20721.427641] input: ckb1: Corsair K95 RGB Gaming Keyboard as /devices/virtual/input/input120
22.08.2016 20:16:48 Mainfrix    kernel  [20721.427859] input: ckb1: Corsair K95 RGB Gaming Keyboard as /devices/virtual/input/input121`
# Running the ckb-daemon with gdb brings following output:
`(gdb) run
Starting program: /home/lutz/Projekte/ckb/bin/ckb-daemon
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
    ckb: Corsair RGB driver beta-v0.2.6+t04
    [I] Root controller ready at /dev/input/ckb0
    [New Thread 0x7ffff6b5b700 (LWP 30508)]
    [I] Connecting Corsair K95 RGB Gaming Keyboard at /dev/input/ckb1
    [New Thread 0x7ffff635a700 (LWP 30511)]
    [I] Starting input thread for /dev/input/ckb1
    [New Thread 0x7ffff5b59700 (LWP 30512)]
    [I] Setup finished for /dev/input/ckb1

**Here is the crash:**
    # [E] os_usbsend (via led_keyboard.c:131): No such device
    [I] Attempting reset...
    [E] os_resetusb (via usb.c:169): resetusb failed: No such device
    [E] usb_tryreset (usb.c:177): Reset failed. Disconnecting.
    [I] Stopping input thread for /dev/input/ckb1
    [I] Disconnecting /dev/input/ckb1
    [Thread 0x7ffff635a700 (LWP 30511) exited]
    [I] Removed device path /dev/input/ckb1
    [Thread 0x7ffff5b59700 (LWP 30512) exited]
    [Thread 0x7ffff6b5b700 (LWP 30508) exited]
    [New Thread 0x7ffff5b59700 (LWP 2684)]
    [I] Connecting Corsair K95 RGB Gaming Keyboard at /dev/input/ckb1
    [New Thread 0x7ffff635a700 (LWP 2702)]
    [I] Starting input thread for /dev/input/ckb1
    [New Thread 0x7ffff5154700 (LWP 2703)]
    [I] Setup finished for /dev/input/ckb1
`
**Do you have the same effect?**
You can take my 6f9331a (is called macrotime.0.2) to check this behaviour.
BTW: In my environment UINT_MAX was undefined, so I needed to include a headerfile.
Here you can find a more detailed output:
[protocol_ckb.txt](https://github.com/ccMSC/ckb/files/431085/protocol_ckb.txt)
cu. Lutz